source/AI.cpp

```diff
@@ -1715,7 +1715,28 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, const System *playerSy
 	if(ship.Health() < minHealth && (!ship.IsYours() || Preferences::Has("Damaged fighters retreat")))
 		return true;
 
-	// TODO: Reboard if in need of ammo.
```

Comment:
I tested a version of this PR (with a recent master-branch merged into it) and noticed that the Outfitter feature "Do you want to reload all the ammunition for your ships?" didn't work for automatically reloading Ammo storage racks for ships that don't also have the proper weapon to fire such Ammo installed. (Which could be the case for carriers that want to resupply their fighters.)

Reply:
That would certainly be a helpful addition with this

Reply:
I believe that is added now

Reply:
Yes, the added code seems to resolve this.